src/status_im2/contexts/wallet/common/utils.cljs

```diff
@@ -23,23 +24,23 @@
   (let [path (get-derivation-path number-of-accounts)]
     (format-derivation-path path)))
 
-(defn- calculate-raw-balance
-  [raw-balance decimals]
-  (if-let [n (utils.number/parse-int raw-balance nil)]
-    (/ n (Math/pow 10 (utils.number/parse-int decimals)))
-    0))
-
-(defn- total-token-value-in-all-chains
-  [{:keys [balances-per-chain decimals]}]
+(defn- total-raw-balance-in-all-chains
+  [balances-per-chain]
   (->> balances-per-chain
-       (vals)
-       (map #(calculate-raw-balance (:raw-balance %) decimals))
-       (reduce +)))
+       (map (comp :raw-balance val))
+       (reduce money/add)))
+
+(defn- total-token-fiat-value
+  "Returns the total token fiat value taking into account all token's chains."
+  [{:keys [balances-per-chain decimals market-values-per-currency]}]
+  (let [usd-price                 (-> market-values-per-currency :usd :price)
+        total-units-in-all-chains (-> balances-per-chain
+                                      (total-raw-balance-in-all-chains)
+                                      (money/token->unit decimals))]
+    (money/crypto->fiat total-units-in-all-chains usd-price)))
 
-(defn calculate-balance
-  [tokens-in-account]
-  (->> tokens-in-account
-       (map (fn [token]
-              (* (total-token-value-in-all-chains token)
-                 (-> token :market-values-per-currency :usd :price))))
```

Comment:
`utils.money/crypto->fiat` does the same but supports bignumbers, so not it's being used